### PR TITLE
Add a cdeps test and update mom6, fv3, and weekly tests

### LINKED ISSUE #504: Update upp submodule for fixing UPHL issue found in 3-km RRFS North American grid
## Description
There is is an issue found from inline post for UPHL 2-5km issue in 3-km RRFS North American grid

## Solution
Update upp submodule with the latest UPP release upp_v10.0.12.

## Alternatives
If applicable, add a description of any alternative solutions or features you've considered.

## Testing:

1. Have you tested the code changes? On what platforms? Yes, on WCOSS1-Dell
3. Have you run regression test in ufs-weather-model or ufs-s2s-model with code changes? Yes.
- Will the baseline results change? Yes
- If yes, please give brief explanation of your code updates on the regression test.
  There might be updates from inline post output grib2 files

## Dependent PRs:

Directly reference any issues or PRs in this or other repositories that this is related to, and describe how they are related. Example:
- required to support noaa-emc/GFDL_atmos_cubed_sphere/issues/<issue_number>: No
- ncar/ccpp-physics/pull/<pr_number>: No
- associated ufs-weather-model/pull/<pr_number>: No

### LINKED ISSUE #1101: Set up weekly tests on Jet to run three coupled p8 tests
## Description
Three coupled p8 tests will be set up to run weekly on Jet.

## Solution
The rt_weekly.conf file will be revised to add the following three coupled p8 tests:
1. cpld_control_c192_p8
2. cpd_control_c384_p8
3. cpld_bmark_p8

A cron job will be set up to run weekly on Jet.

### LINKED ISSUE #1126: Update upp submodule for fixing UPHL issue found in 3-km RRFS North American grid
## Description
There is is an issue found from inline post for UPHL 2-5km issue in 3-km RRFS North American grid.

## Solution
Update upp submodule with the latest UPP release upp_v10.0.12.

## Alternatives
None.

## Related to
Directly reference any issues or PRs in this or other repositories that this is related to, and describe how they are related. Example:
- [FV3 Issue #504](https://github.com/NOAA-EMC/fv3atm/issues/504)
- [FV3 PR #505](https://github.com/NOAA-EMC/fv3atm/pull/505)

### LINKED ISSUE #1128: update MOM6 to its main branch 20220317 commit
MOM6 main branch was updated on 20220317, see detail at https://github.com/mom-ocean/MOM6/pull/1559. Need to update dev/emc correspondingly.

### PR DESCRIPTION
**Pull request by binli2337**
# PR Checklist

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [x] Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.

- [ ] New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsiblity to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

This is a combined pull request. It includes updates from PR #1127 , PR #1129 and PR #1132 .

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!) 

- fixes [#1126](https://github.com/ufs-community/ufs-weather-model/issues/1126) 
- fixes [noaa-emc/fv3atm/issues/#504](https://github.com/NOAA-EMC/fv3atm/issues/504)

- fixes #1128 
- MOM6 issue https://github.com/NOAA-EMC/MOM6/issues/86

- fixes #1101 

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [ ] opnReqTest for newly added/changed feature
- [x] CI

## Dependencies

If testing this branch requires non-default branches in other repositories, list them. Those branches should have matching names (ideally).

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs

- waiting on [noaa-emc/fv3atm/pull/<505>](https://github.com/NOAA-EMC/fv3atm/pull/505)

Co-authored-by: @WenMeng-NOAA <wen.meng@noaa.gov>
Co-authored-by: @jiandewang <jiande.wang@noaa.gov>
Co-authored-by: @DusanJovic-NOAA  <dusan.jovic@noaa.gov>

### CODE REVIEW

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: cheyenne
Compiler: gnu
Job: BL
[BL] Repo location: /glade/scratch/dtcufsrt/autort/tests/auto/pr/886391713/20220322163013/ufs-weather-model
[BL] Error: Test compile_007 failed in run_compile failed 
Please make changes and add the following label back: cheyenne-gnu-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: jet
Compiler: intel
Job: RT
[RT] Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/886391713/20220322234509/ufs-weather-model
Please make changes and add the following label back: jet-intel-RT

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: gaea
Compiler: intel
Job: BL
[BL] Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/886391713/20220322223007/ufs-weather-model
[BL] Error: Test cpld_control_c384_p8 007 failed in run_test failed 
[BL] Error: Test control_c384 015 failed in run_test failed 
[BL] Error: Test control_c384gdas 016 failed in run_test failed 
[BL] Error: Test control_p7_rrtmgp 022 failed in run_test failed 
[BL] Error: Test control_ras 042 failed in run_test failed 
[BL] Error: Test control_thompson 043 failed in run_test failed 
[BL] Error: Test control_CubedSphereGrid_debug 047 failed in run_test failed 
[BL] Error: Test control_stochy_debug 049 failed in run_test failed 
[BL] Error: Test control_lndp_debug 050 failed in run_test failed 
[BL] Error: Test control_rrtmgp_debug 051 failed in run_test failed 
[BL] Error: Test control_ras_debug 054 failed in run_test failed 
[BL] Error: Test control_thompson_extdiag_debug 059 failed in run_test failed 
[BL] Error: Test rap_lndp_debug 065 failed in run_test failed 
Please make changes and add the following label back: gaea-intel-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: cheyenne
Compiler: gnu
Job: BL
[BL] Repo location: /glade/scratch/dtcufsrt/autort/tests/auto/pr/886391713/20220323044513/ufs-weather-model
[BL] Error: Test compile_007 failed in run_compile failed 
Please make changes and add the following label back: cheyenne-gnu-BL

**Comment by binli2337** [comment]
Errors from control_p7_rrtmgp on Gaea:

 11: fv3.exe: lib/darshan-common.c:262: darshan_track_common_val_counters: Assertion `found == counter' failed.
 11: forrtl: error (76): Abort trap signal

**Comment by DeniseWorthen** [comment]
@jiandewang could you document which of the combined PRs here cause the baseline change? MOM6 updates don't usually change baselines so it would be good add that information.

**Comment by jiandewang** [comment]
@DeniseWorthen this MOM PR doesn't change answer. The original MOM6 PR us at https://github.com/ufs-community/ufs-weather-model/pull/1129.

**Comment by jiandewang** [comment]
@binli2337 and @WenMeng-NOAA can you add some information here regarding the baseline changes ?

**Comment by WenMeng-NOAA** [comment]
Regarding to upp submodule updates at FV3 component, there would be changes of datasets PRSLEV.GrbF/NATLEV.GrbF from regional FV3 inline post in RT test baselines.

**Comment by jiandewang** [comment]
@binli2337 I just merged MOM6, you can switch UFS submodule by the time, thanks for your time and effect

**Review by jiandewang — APPROVED** [review]

File: tests/tests/cpld_control_c192_p8 [hunk 1]
```diff
@@ -56,6 +56,7 @@ export LIST_FILES="sfcf036.tile1.nc \
 
 export_fv3
 export_cpl
+export WLCLK=48
```
Comment (junwang-noaa) [inline tests/tests/cpld_control_c192_p8:59]: Why do you need to increase the WLCLK time?

Comment (binli2337) [reply tests/tests/cpld_control_c192_p8:59]: This test needs about 43 minutes to run on Jet.

Comment (junwang-noaa) [reply tests/tests/cpld_control_c192_p8:59]: Since this is for weekly test. We will remove this line in next PR and have a separate fix in weekly test.

**Review by junwang-noaa — APPROVED** [review]

**Comment by junwang-noaa** [comment]
@binli2337 fv3 PR is merged, revision: [55e98b2](https://github.com/NOAA-EMC/fv3atm/commit/55e98b2986847406a8b4fdf6c578fd4e35765a67)

**Pull request merged**